### PR TITLE
Added Option to navigate in the html documents

### PR DESCRIPTION
**Pull request by RamolaWeb**
In this I have created  Article  class to hold the title and content of each article extracted and then used this title to navigate in the html documents. This is to fix issue #12

### CODE REVIEW

**Comment by RamolaWeb** [comment]
@CuriousLearner  Please have a review on it.

File: Article.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Article(object):
```
Comment (CuriousLearner) [inline Article.py]: Please use module name as `article`. So import becomes `from article import Article`

File: g4g.py [hunk 2]
```diff
@@ -45,13 +46,21 @@ def save_articles_as_html_and_pdf():
                     '</head><body>'
                     )
     allArticles += '<h1 style="text-align:center;font-size:40px">' + categoryUrl.title() + ' Archive</h1><hr>'
-    allArticles += '<hr>'.join(articles)
+    allArticles += '<h1 style="text-align:center;font-size:40px">Content</h1>'
+    i=1
+    for x in range(len(articles)):
+        allArticles+='<a href =\"#'+str(i)+'\">'+'<h1 style="font-size:20px;">'+ str(i) + ".\t\t"+articles[x].title + '</h1></a> <br/>'
```
Comment (CuriousLearner) [inline g4g.py]: Please follow PEP8.

It would make code more readable.

File: g4g.py [hunk 3]
```diff
@@ -45,13 +46,21 @@ def save_articles_as_html_and_pdf():
...
+        i=i+1
+    j=1
+    for x in range(len(articles)):
+        allArticles+='<hr id=\"'+str(j)+'\">'+articles[x].content.decode("utf-8")
```
Comment (CuriousLearner) [inline g4g.py]: PEP8 here also.

File: g4g.py [hunk 4]
```diff
@@ -88,7 +97,8 @@ def scrape_category(categoryUrl):
                 code_tag['class'] = code_tag.get('class', []) + ['prettyprint']
             article = link_soup.find('article')
             # Now add this article to list of all articles
-            articles.append(article.encode('UTF-8'))
+            page=Article(title=link_soup.title.string,content=article.encode('UTF-8'))
```
Comment (CuriousLearner) [inline g4g.py]: PEP8, leave  a space after and before `=`, `,`.

**Comment by RamolaWeb** [comment]
@CuriousLearner  Fix Code According to PEP8.Please Review it

File: article.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Article(object):
+    """This Class Contains the title and the content of the article.The title
+     can be use as key to the article link for the navigation purpose ."""
+    def __init__(self, title, content):
```
Comment (CuriousLearner) [inline article.py]: Leave a line after docstring. Also there should be a space after full-stop.

File: g4g.py [hunk 6]
```diff
@@ -45,13 +46,21 @@ def save_articles_as_html_and_pdf():
                     '</head><body>'
                     )
     allArticles += '<h1 style="text-align:center;font-size:40px">' + categoryUrl.title() + ' Archive</h1><hr>'
-    allArticles += '<hr>'.join(articles)
+    allArticles += '<h1 style="text-align:center;font-size:40px">Content</h1>'
```
Comment (CuriousLearner) [inline g4g.py]: Rather than `Content`, `Index` or something similar would look better here.

**Comment by CuriousLearner** [comment]
@RamolaWeb I've taken a pass, in case you've any doubts, don't hesitate to ask here!

Thanks again for your contributions.

**Comment by RamolaWeb** [comment]
@CuriousLearner  Fix codes according to review and change layout of the html page. But cannot able to render the links in the pdf.Please review it.

File: g4g.py [hunk 7]
```diff
@@ -46,10 +46,10 @@ def save_articles_as_html_and_pdf():
                     '</head><body>'
                     )
     allArticles += '<h1 style="text-align:center;font-size:40px">' + categoryUrl.title() + ' Archive</h1><hr>'
-    allArticles += '<h1 style="text-align:center;font-size:40px">Content</h1>'
+    allArticles += '<h1 style="padding:0 5% 0 5%;font-size:200%;"> Index</h1><Br/>'
     i = 1
     for x in range(len(articles)):
-        allArticles += '<a href =\"#'+str(i)+'\">'+'<h1 style="font-size:20px;">' + str(i) + ".\t\t"+articles[x].title + '</h1></a> <br/>'
+        allArticles += '<a href =\"#'+str(i)+'\">'+'<h1 style="padding:0 5% 0 5%;font-size:100%;">' + str(i) + ".\t\t"+articles[x].title + '</h1></a> <br/>'
```
Comment (CuriousLearner) [inline g4g.py]: Can we do better here rather than giving padding this way?

Comment (CuriousLearner) [inline g4g.py]: Also, why is font-size changed?

**Comment by RamolaWeb** [comment]
@CuriousLearner fix padding and font size of the index section.Please comment and review this.

**Comment by CuriousLearner** [comment]
Hi @RamolaWeb No need of changes directly, first we can discuss and then fix issues. That way we can collaborate better.

So, in the earlier review, "Can we do better here rather than giving padding this way?", I was trying to refer to my earlier review of using CSS class. 

There is a lot of code that needs to be refactored in the project.

Plus, I will try to figure out the issue for the PDF links and have to fix that before we get this PR merged in.

So, let's hold on this PR for sometime, and I'll update here with my findings.

Thanks again for your contributions :)

**Comment by RamolaWeb** [comment]
@CuriousLearner  Thanks for reviewing the pr and i will contribute to other issue in the project.

**Comment by CuriousLearner** [comment]
@RamolaWeb It was written around 2 years ago, and it needs a lot of refactoring, plus better code, I will try to arrange it in directory structure and do that soon. We can then figure out if more features are needed.

**Comment by CuriousLearner** [comment]
Looks to me like the issue is with different versions of `wkhtmltopdf`. I've to still verify this on different OS.

If that is the case, we can merge this in, and figure out later on how to make links work in the pdf versions.

**Comment by RamolaWeb** [comment]
It is a problem with the wkhtmlpdf as mention in the https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2153 . The Internal Links are not working properly in differernt os as mentioned.

**Comment by CuriousLearner** [comment]
@RamolaWeb Looks good. There are some nits, which I'll fix and then merge this.

Thanks a lot for your work.

**Comment by CuriousLearner** [comment]
Merged in #14 

Thanks for your work Sahil
@RamolaWeb

**Comment by RamolaWeb** [comment]
@CuriousLearner Thanks for merging PR and I will give more contribution in this Project

**Comment by CuriousLearner** [comment]
Sure @RamolaWeb !

I've started cleaning the code and would maintain a proper structure for the Project. It's on #15

**Comment by RamolaWeb** [comment]
@CuriousLearner  I Think You Have Closed My PR due to which my own fork repository is showing that it is ahead of Your Repo by 6 Commit. What Should I do?

**Comment by CuriousLearner** [comment]
@RamolaWeb Lessons to learn:

- Your PR should not be from your master branch. It should be from a feature branch on your fork. In case an organization/project uses squash or as in this case separate PR for adding more work, your branch would be outdated with upstream.

- What to do now?
   - Make sure you preserve your changes in  a separate branch on your fork (optional)
   - Do a git reset --hard aa9216d63e16bb4891e27ac64c3e9b87a9cf07de
   - Now pull from upstream like
      git pull git@github.com:CuriousLearner/GeeksForGeeksScrapper.git master

**Comment by CuriousLearner** [comment]
@RamolaWeb Did it resolve the issue? Let me know if you face any problem.

**Pull request closed without merging**